fbgemm_gpu/include/fbgemm_gpu/fbgemm_cuda_utils.cuh

```diff
@@ -1212,6 +1216,9 @@ DEVICE_INLINE float_16 make_zero_float_16() {
 
 __forceinline__ __device__ __half2
 hfma2(const __half2 a, const __half2 b, const __half2 c) {
+#ifdef __HIP_PLATFORM_HCC__
+  return __hfma2(a, b, c);
+#else
```

Comment:
Can I ask you to merge the condition? For example,
```
#if (__CUDA_ARCH__ >= 530 && __CUDA_ARCH__ != 610) || defined(__HIP_PLATFORM_HCC__)
...
```

It'd be nice if you can apply this change to other changes below.

Reply:
Thank you for your suggestions. The conditions are combined.